cleanup.go

```diff
@@ -2,35 +2,37 @@ package testcontainers
 
 import (
 	"context"
-	"errors"
 	"fmt"
 	"reflect"
 	"time"
 )
 
-// terminateOptions is a type that holds the options for terminating a container.
-type terminateOptions struct {
-	ctx     context.Context
-	timeout *time.Duration
-	volumes []string
+// DefaultTimeout for termination
+var DefaultTimeout = 10 * time.Second
+
+// TerminateOptions is a type that holds the options for terminating a container.
+type TerminateOptions struct {
+	Context context.Context
+	Timeout *time.Duration
+	Volumes []string
```

Comment:
suggestion: use a constructor and methods to ensure consistent behaviour with minimum effort from different types.

```suggestion
// TerminateOptions is a type that holds the options for terminating a container.
type TerminateOptions struct {
        stopTimeout *time.Duration
        volumes []string
        ctx context.Context
}

// NewTerminateOptions returns a fully initialised TerminateOptions.
// Defaults: StopTimeout: 10 seconds.
func NewTerminateOptions(ctx context.Context, opts ...TerminateOption) *TerminateOptions {
        stopTimeout := 10 * time.Second
	options := &TerminateOptions{
		stopTimeout: &stopTimeout,
		ctx: ctx,
	}

	for _, opt := range opts {
		opt(options)
	}

	return options
}

// Context returns the context to use duration a Terminate.
func (o *TerminateOptions) Context() context.Context {
        return o.ctx
}

// StopTimeout returns the stop timeout to use duration a Terminate.
func (o *TerminateOptions) StopTimeout() time.Duration {
        return o.stopTimeout
}

// Cleanup performs any clean up needed.
func (o *TerminateOptions) Cleanup() error {
	// TODO: simplify this when when perform the client refactor.
	if len(o.volumes) == 0 {
		return nil
	}

	client, err := NewDockerClientWithOpts(o.ctx)
	if err != nil {
		return fmt.Errorf("docker client: %w", err)
	}
	defer client.Close()

	// Best effort to remove all volumes.
	var err []errors
	for _, volume := range o.volumes {
		if errRemove := client.VolumeRemove(o.ctx, volume, true); errRemove != nil {
			errs = append(errs, fmt.Errorf("volume remove %q: %w", volume, errRemove))
		}
	}

	return errors.Join(errs...)